test/suites/consolidation/suite_test.go

```diff
@@ -222,4 +225,104 @@ var _ = Describe("Consolidation", func() {
 
 		env.ExpectDeleted(largeDep, smallDep)
 	})
+	It("should consolidate on-demand nodes to spot (replace)", func() {
+		provider := test.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{AWS: awsv1alpha1.AWS{
+			SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": env.ClusterName},
+			SubnetSelector:        map[string]string{"karpenter.sh/discovery": env.ClusterName},
+		}})
+		provisioner := test.Provisioner(test.ProvisionerOptions{
+			Requirements: []v1.NodeSelectorRequirement{
+				{
+					Key:      v1alpha5.LabelCapacityType,
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{"on-demand"},
+				},
+				{
+					Key:      v1.LabelInstanceTypeStable,
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{"c4.large"},
```

Comment:
Can we leave the instance type open here or maybe just add an instance-size requirement for "large"?